### PR TITLE
Doc linkcode tuning

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
make the `[source]` anchors point to the correct file on github wherever reasonably possible, including the correct line numbers and the correct blob. I chose the output of `git describe --tags` here as it

* makes sure that on dev branches we don't point to master
* makes sure to point directly to the corresponding tag when building from a release commit

A build of this branch is up at https://python-telegram-bot.readthedocs.io/en/doc-linkcode-tuning/

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -397,23 +399,91 @@ def autodoc_skip_member(app, what, name, obj, skip, options):
...
+
+
+git_branch = _git_branch()
+base_url = "https://github.com/python-telegram-bot/python-telegram-bot/blob/"
+
+
+def linkcode_resolve(_, info):
+    """See www.sphinx-doc.org/en/master/usage/extensions/linkcode.html"""
+    combined = '.'.join((info['module'], info['fullname']))
+    # special casing for ExtBot which is due to the special structure of extbot.rst
```
Comment (harshil21) [inline docs/source/conf.py:461]: ah mystery solved 👍🏼

File: telegram/ext/_extbot.py [hunk 2]
```diff
@@ -202,7 +202,7 @@ def insert_callback_data(self, update: Update) -> None:
             *In place*, i.e. the passed :class:`telegram.Message` will be changed!
 
         Args:
-            update (:class`telegram.Update`): The update.
+            update (:class:`telegram.Update`): The update.
```
Comment (harshil21) [inline telegram/ext/_extbot.py:205]: ah this was in the referencing pr so I'd left it

**Review by harshil21 — APPROVED** [review]
Nice work! I definitely wouldn't have thought of using an unrelated function to achieve this.

File: telegram/ext/_extbot.py (hunk 2 above)
Comment (Bibo-Joshi) [reply telegram/ext/_extbot.py:205]: ah yes, didn't think of that :D shouldn't hurt, though …

**Pull request merged**